test/e2e_node/device_plugin_test.go

```diff
@@ -428,6 +465,56 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 			framework.ExpectNoError(err, "inconsistent device assignment after pod restart")
 		})
 
+		ginkgo.It("will not attempt to admit the succeeded pod after the kubelet restart and device plugin removed", func(ctx context.Context) {
+			podRECMD := fmt.Sprintf("devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep %s", sleepIntervalWithRestart)
+			podSpec := makeBusyboxPod(SampleDeviceResourceName, podRECMD)
+			podSpec.Spec.RestartPolicy = v1.RestartPolicyNever
+			podSpec.ObjectMeta.Finalizers = []string{testFinalizer}
```

Comment:
Just curious -- why do we need a finalizer here? What will delete the pod if finalizer is not set?

Reply:
+1 to the question, thanks!

Reply:
I have added a comment. Not sure why it is cleaning up fast in my set up. With finalizers it is guarantee to hang around